main.cpp

```diff
@@ -402,3 +403,157 @@ int main(int argc, char** argv) {
 
     return 0;
 }
+else {
+    //User wants to Restore to Stock
+    //Check if Backup Exists
+    RenderMenuLoading(task_percent(task++), "Checking for Backup... (wave.rpx)");
+    PresentMenu();
```

Comment:
If all we're doing is an `if` statement, there's no need to render the whole menu out since the actual check will complete basically instantly. Updating the loading screen makes sense for things like sha hashing or copying files that actually take a fair bit of time.
Perhaps this entire check should be moved up to the confirmation screen? That needs a redesign of its own though (see #3) so it's probably fine for now.